design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
```

Comment:
I'm on board with rich output, but doubt the value of configurability, at least as a main goal. I think it adds significant overhead to implementation, testing and documentation. I'd prefer us focusing our energy on a great default that is in line with our brand before we tackle configurability.